examples/pytorch/deepspeed-demo/train_bert_ds.py

```diff
@@ -0,0 +1,829 @@
+        )
+
+        pad_token_id = self.config.pad_token_id
+        # (labels have also been padded with pad_token_id)
+        # filter out all masked labels
+        # shape: (num_masked_tokens,)
+        masked_token_indexes = torch.nonzero(
+            (tgt_tokens != pad_token_id).view(-1)
+        ).view(-1)
+        # shape: (num_masked_tokens, vocab_size)
+        prediction_scores = self.lm_head(sequence_output, masked_token_indexes)
+        # shape: (num_masked_tokens,)
+        target = torch.index_select(tgt_tokens.view(-1), 0, masked_token_indexes)
+
+        loss_fct = nn.CrossEntropyLoss(ignore_index=-1)
+
+        masked_lm_loss = loss_fct(
+            prediction_scores.view(-1, self.config.vocab_size), target
+        )
+        return masked_lm_loss
+
+
+def create_model(
+    num_layers: int, num_heads: int, ff_dim: int, h_dim: int, dropout: float
+) -> RobertaMLMModel:
+    """Create a Bert model with the specified `num_heads`, `ff_dim`,
+    `h_dim` and `dropout`
+
+    Args:
+        num_layers (int):
+            The number of layers
+        num_heads (int):
+            The number of attention heads
+        ff_dim (int):
+            The intermediate hidden size of
+            the feed forward block of the
+            transformer
+        h_dim (int):
+            The hidden dim of the intermediate
+            representations of the transformer
+        dropout (float):
+            The value of dropout to be used.
+            Note that we apply the same dropout
+            to both the attention layers and the
+            FF layers
+
+    Returns:
+        RobertaMLMModel:
+            A Roberta model for MLM task
+
+    """
+    roberta_config_dict = {
+        "attention_probs_dropout_prob": dropout,
+        "bos_token_id": 0,
+        "eos_token_id": 2,
+        "hidden_act": "gelu",
+        "hidden_dropout_prob": dropout,
+        "hidden_size": h_dim,
+        "initializer_range": 0.02,
+        "intermediate_size": ff_dim,
+        "layer_norm_eps": 1e-05,
+        "max_position_embeddings": 514,
+        "model_type": "roberta",
+        "num_attention_heads": num_heads,
+        "num_hidden_layers": num_layers,
+        "pad_token_id": 1,
+        "type_vocab_size": 1,
+        "vocab_size": 50265,
+    }
+    roberta_config = RobertaConfig.from_dict(roberta_config_dict)
+    roberta_encoder = RobertaModel(roberta_config)
+    roberta_model = RobertaMLMModel(roberta_config, roberta_encoder)
+    return roberta_model
+
+
+# Experiment Management Related Functions
+
+
+def get_unique_identifier(length: int = 8) -> str:
+    """Create a unique identifier by choosing `length`
+    random characters from list of ascii characters and numbers
+    """
+    alphabet = string.ascii_lowercase + string.digits
+    uuid = "".join(alphabet[ix] for ix in np.random.choice(len(alphabet), length))
+    return uuid
+
+
+def create_experiment_dir(
```

Comment:
Do we need this experiment dir in this example ?

Reply:
Similar to the issue above, it will create a directory in the checkpoint_dir of rank 0, I think we can stay consistent with DeepSpeedExamples.